src/bin/luster.rs

```diff
@@ -114,6 +120,21 @@ fn main() -> Result<(), Box<StdError>> {
 
     let file = io::buffered_read(File::open(matches.value_of("file").unwrap())?)?;
 
+    if matches.is_present("dump") {
+        lua.sequence(|root| {
+            sequence::from_fn_with(root, |mc, root| {
+                let proto = compile(mc, root.interned_strings, file)?;
+                println!("{}", proto);
+                Ok(())
+            })
+            .map_ok(|_| ())
+            .map_err(|e: Error| e.to_static())
+            .boxed()
+        })?;
+        return Ok(());
+    }
+
+    let file = io::buffered_read(File::open(matches.value_of("file").unwrap())?)?;
```

Comment:
This is reading the file argument twice if not given the "dump" flag, maybe an `else` is appropriate here?  Also, it doesn't seem to complain if both `--dump` and `--repl` are given, even though it won't run the repl, though that might be fine for the moment.

Reply:
Thanks for the review!

I've removed the redundant read in https://github.com/kyren/luster/pull/14/commits/f92af5bba2616a5e61f77c928926b6055efc1116.

https://github.com/kyren/luster/pull/14/commits/e87af933231846a35ee803bac234c77d2ea73967 makes luster return an error if both `--dump` and `--repl` are given.